### PR TITLE
Update: enable logging via config

### PR DESCRIPTION
**Pull request by till**
Hopefully didn't change much and kept all BC. The idea is to allow
the terraform openstack provider to enable logging using the
configuration struct, vs. using environment variables.

Related:
terraform-provider-openstack/terraform-provider-openstack#1246
terraform-provider-openstack/terraform-provider-openstack#1203

### CODE REVIEW

File: terraform/auth/config.go [hunk 1]
```diff
@@ -425,3 +426,18 @@ func (c *Config) SharedfilesystemV2Client(region string) (*gophercloud.ServiceCl
 func (c *Config) KeyManagerV1Client(region string) (*gophercloud.ServiceClient, error) {
 	return c.CommonServiceClientInit(openstack.NewKeyManagerV1, region, "key-manager")
 }
+
+// A wrapper to determine if logging in gophercloud should be enabled, with a fallback
+// to the OS_DEBUG environment variable when no explicit configuration is passed.
+func enableLogging(enable bool) bool {
```
Comment (till) [inline terraform/auth/config.go:432]: Figured I make this a _private_ utility function, e.g. not part of the config class. But let me know if that's wanted or if you prefer this differently.

File: terraform/auth/config.go [hunk 2]
```diff
@@ -59,6 +59,7 @@ type Config struct {
 
 	TerraformVersion string
 	SDKVersion       string
+	EnableLogger     bool
```
Comment (till) [inline terraform/auth/config.go:62]: Put this next to other terraformy stuff, as it's not really an openstack thing.

**Review by till — COMMENTED** [review]
Left two comments to explain what I was thinking. Appreciate all feedback.

**Comment by jtopjian** [comment]
@kayrus @ozerovandrei This looks good to me and it looks like it's backwards compatible. Can one of you two review when you have a moment?

**Comment by till** [comment]
@ozerovandrei can I trouble you to look into this?

**Review by ozerovandrei — APPROVED** [review]
LGTM
Thank you @till

**Comment by till** [comment]
@jtopjian Thank you!

**Pull request merged**